src/datasource.tsx

```diff
@@ -1016,12 +1015,12 @@ export class PrometheusDatasource
     delete variables.__interval_ms;
 
     //Add ad hoc filters
-    const expr = this.enhanceExprWithAdHocFilters(target.expr);
+    const expr = this.templateSrv.replace(target.expr, variables, this.interpolateQueryExpr);
 
     return {
       ...target,
       legendFormat: this.templateSrv.replace(target.legendFormat, variables),
-      expr: this.templateSrv.replace(expr, variables, this.interpolateQueryExpr),
```

Comment:
hi @yincongcyincong ! can you add the tests for your changes?

Reply:
ok , i will add

Reply:
@yincongcyincong, it would be great if you could add a CHANGELOG description about the fix. Thank you!

Reply:
@dmitryk-dk I think we can add changelog line on our own if the follow-up commit, since @yincongcyincong might not have a context how we publish changes.

Reply:
i added the unit test, please check it.